### DISCUSSION TITLE
How to find all possible way to combine sum with python

### DISCUSSION THREAD

**Discussion by Kitsune-cop** [community/community#75975] [Programming Help]
### Select Topic Area

Question

### Body

Given a list of numbers and a value,return the number of ways to generate the value using the numbers in the list. You can use a number multiple times. Don't use any built-in functions or library

```
Example:
list = [1,2,3]
target = 4

expected result: [[1,1,1,1], [1,1,2], [1,2,1], [2,1,1], [2,2], [1,3], [3+1]], 7 ways
```

I have backtrack code like this:

```
def combinationSum(candidates, target):
    result = []
    def backtrack(pos, target, currentCombination):
        if target == 0:
            print("new combination", currentCombination)
            result.append(currentCombination[::])
            return    
        if pos == len(candidates) or target < 0:
            return    
        for i in range(pos, len(candidates)):
            currentElement = candidates[i]
            currentCombination.append(currentElement)
            backtrack(i, target-currentElement, currentCombination)
            currentCombination.pop()        
    backtrack(0, target, [])   
    return result

Output is : [[1,1,1,1], [1,1,2], [1,3], [2,2] it print only 4 ways

```
Please help me fix that code and print the result same expect result

**Comment by EstebanCRz** [comment]
you should pass "i" instead of "pos" in the recursive call to ensure that you can reuse the same number in the combination.
Here's the modified code:
def combinationSum(candidates, target):
    result = []
    def backtrack(pos, target, currentCombination):
        if target == 0:
            print("new combination", currentCombination)
            result.append(currentCombination[::])
            return    
        if pos == len(candidates) or target < 0:
            return    
        for i in range(pos, len(candidates)):
            currentElement = candidates[i]
            currentCombination.append(currentElement)
            # Pass i instead of pos to allow reusing the same number
            backtrack(i, target - currentElement, currentCombination)
            currentCombination.pop()        
    backtrack(0, target, [])   
    return result

# Test with the provided example
my_list = [1, 2, 3]
my_target = 4
result = combinationSum(my_list, my_target)
print(result)


result = [[1, 1, 1, 1], [1, 1, 2], [1, 2, 1], [2, 1, 1], [2, 2], [1, 3], [3, 1]]

**Comment by queenofcorgis** [comment]
Hi @Kitsune-cop ,
The GitHub Community on Discussions is a space for all GitHub users to come and discuss ideas and projects. Your post was flagged and close because it violated our [Community Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md), we ask that members contribute in a positive and constructive way. Posting homework or test questions without context will likely discourage engagement and responses.

If this content is posted again, your Community account will be subject to immediate and indefinite suspension.
For additional guidance, please review our [Community Code of Conduct](https://github.com/community/community/blob/main/CODE_OF_CONDUCT.md) and [GitHub’s Terms of Service](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service).

Thank you!
